x-pack/plugins/osquery/cypress/e2e/roles/reader.cy.ts

```diff
@@ -76,31 +72,23 @@ describe('Reader - only READ', { tags: ['@ess'] }, () => {
     navigateTo('/app/osquery/live_queries');
     cy.contains('New live query').should('be.disabled');
     cy.contains(liveQueryQuery);
-    cy.react('EuiIconPlay', { options: { timeout: 3000 } }).should('not.exist');
-    cy.react('ActionTableResultsButton').should('exist');
+    cy.get(customActionRunSavedQuerySelector(savedQueryName)).should('not.exist');
+    cy.get(`[aria-label="Details"]`).should('exist');
   });
 
   it('should not be able to add nor edit packs', () => {
     navigateTo('/app/osquery/packs');
     cy.contains('Add pack').should('be.disabled');
     cy.getBySel('tablePaginationPopoverButton').click();
     cy.getBySel('tablePagination-50-rows').click();
-    cy.react('ActiveStateSwitchComponent', {
-      props: { item: { name: packName } },
-    })
-      .find('button')
-      .should('be.disabled');
+
+    // activate status switch
+    cy.get(`[aria-label="${packName}"]`).should('be.disabled');
```

Comment:
typo: `ActiveStateSwitchComponent` instead of `activate status switch`

but I could suggest a helper constant, too:
```ts
const activeStateSwitchComponentSelector = `[aria-label="${packName}"]`;
cy.get(activeStateSwitchComponentSelector).should('be.disabled');
```